crates/interpreter/src/valid.rs

```diff
@@ -826,3 +898,67 @@ impl<'a, 'm> Expr<'a, 'm> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use alloc::format;
+
+    use proptest::prelude::*;
+
+    use super::bit_field::*;
+    use crate::Error;
+
+    proptest! {
+        #[test]
+        fn signed_field_round_trip(value in -8i32..8) {
+            let mask = 0b1111000;
+            let field = into_signed_field(mask, value).unwrap();
+            let recovered = from_signed_field(mask, field);
+            prop_assert_eq!(value, recovered);
+        }
+
+        #[test]
+        fn unsigned_field_round_trip(value in 0u32..8) {
+            let mask = 0b111000;
+            let field = into_field(mask, value).unwrap();
+            let recovered = from_field(mask, field);
+            prop_assert_eq!(value, recovered);
+        }
+    }
+
+    #[test]
+    #[cfg(debug_assertions)]
+    fn into_field_error_debug() {
+        // This mask has 3 bits set, so it can only represent values 0-7.
+        let mask = 0b111000;
+        let value = 8;
+
+        let result = into_field(mask, value);
+        assert!(result.is_err());
+        assert!(matches!(result.unwrap_err(), Error::Unsupported(_)));
+    }
+
+    #[test]
+    #[cfg(not(debug_assertions))]
+    fn into_field_error_release() {
+        // This mask has 3 bits set, so it can only represent values 0-7.
+        let mask = 0b111000;
+        let value = 8;
+
+        let field = into_field(mask, value).unwrap();
+        assert_ne!(value, field);
+    }
+
+    #[test]
+    fn into_signed_field_overflow() {
+        let mask = 0b1111000;
+        let value = i32::MAX;
```

Comment:
I would actually prefer to split this test in 2:
- `into_signed_field_error_positive()` testing `into_signed_field(0b1111000, 8)`
- `into_signed_field_error_negative()` testing `into_signed_field(0b1111000, -9)`

That said, I believe we should always use proptest instead, even for negative testing. Which means we should replace those unit-tests to those prop-tests:
- `unsigned_field_error(value in 8u32..)`
- `signed_field_error_positive(value in 8i32..)`
- `signed_field_error_negative(value in ..-8i32)`